src/mono/wasm/runtime/library_mono.js

```diff
@@ -2252,6 +2359,57 @@ var MonoSupportLib = {
 		};
 		debugger;
 	},
+
+	mono_wasm_asm_loaded: function (assembly_name, assembly_ptr, assembly_len, pdb_ptr, pdb_len) {
+		if (MONO.mono_wasm_runtime_is_ready !== true)
+			return;
+
+		if (MONO._loaded_filenames === undefined) {
+			MONO._loaded_filenames = [];
+			MONO.loaded_files.forEach(file => {
+				try {
+						let url = new URL(file);
+						let filename = url.pathname.split('/').pop();
+						if (filename.length !== 0)
+								MONO._loaded_filenames.push(filename);
+				} catch (e) { }
+			});
+		}
+
+		const assembly_name_str = assembly_name !== 0 ? Module.UTF8ToString(assembly_name) : '';
+		if (assembly_name !== 0) {
+			if ((assembly_name_str.startsWith("System.") || assembly_name_str.startsWith("Microsoft."))
```

Comment:
I don't like this name hack.  driver.c has a linked list of the preloaded assemblies can't we dedup off that by address?

Reply:
@radical made this change so I'll let him speak on the trade-offs here.

Reply:
Chatted with @thaystg about this. She suspects it wasn't intentional. I'll make the change here.

Reply:
I've added a new exported function in driver.c that searches for assemblies already loaded by name and updated the check here to leverage that.

Reply:
Glad to be rid of the hack! 👍